src/utils.js

```diff
@@ -127,6 +127,54 @@ function pick(o, props = []) {
     }, {});
 }
 
+/**
+ * Utility function used to perform a dfs search over a graph, recursively.
+ * @param {Array.<Object>} links - list of links connecting the nodes in a graph.
+ * @param {number} nodeid - id property of the node currently checking neighbors of.
+ * @param {Object} visited - Object keeping track of all the nodes previously visited.
+ * @param {Object} recStack - Object keeping track of recursion stack, containing the back edges.
+ * @returns {boolean} boolean reflecting wether the input nodes and links, results in a cyclic graph.
+ * @memberof utils
+ */
+const checkCyclicUtil = (links, nodeid, visited, recStack) => {
+    if (!visited[nodeid]) {
+        const nodeNeighbors = links.filter(edge => edge.source === nodeid);
+
+        visited[nodeid] = true;
+        recStack[nodeid] = true;
+
+        for (const neighbor of nodeNeighbors) {
+            if (!visited[neighbor.target] && checkCyclicUtil(links, neighbor.target, visited, recStack)) {
```

Comment:
We should refactor this recursive call to an iterative approach, something like this to traverse using a stack:

```javascript
// not tested code, don't sure if even compiles :)
while (stack.length) {
    const node = stack.pop();

    if (!visited[s]) {
        visited[s] = true;
    }

    const neighbors = Object.keys(linksMatrix[node]);

    if (neighbors && neighbors.length) {
        neighbors.reduce((_stack, nId) => {
            if (!visited[nId]) {
                _stack.push(nId);
            }

            return _stack;
        }, stack);
    }
}
```
Of course this + the logic for cyclic verification somewhere in between.